parser/fsimporter.go

```diff
@@ -732,6 +756,188 @@ func (fs *FSImporter) parseFiles(indexedFiles []*fpt.IndexedFile, parsingThreads
+											uconnMap[srcDstKey].ConnStateList[uid].Duration = duration
+
+											// If current duration is longer than previous duration, we can
+											// also assume that current bytes is /at least/ as big as the
+											// stored value for bytes...same for OrigBytes
+											uconnMap[srcDstKey].ConnStateList[uid].Bytes = bytes
+											uconnMap[srcDstKey].ConnStateList[uid].OrigBytes = origIPBytes
+										}
+									}
+								} else {
+									// No entry was present for a connection with this UID. Create a new
+									// entry and set the Open state accordingly
+									uconnMap[srcDstKey].ConnStateList[uid] = &uconn.ConnState{
+										Bytes:     bytes,
+										Duration:  duration,
+										Open:      true,
+										OrigBytes: origIPBytes,
+										Ts:        ts, //ts is the timestamp at which the connection was detected
+										Tuple:     tuple,
+									}
+								}
+
+								// this is to keep track of how many times a host connected to
+								// an unexpected port - proto - service Tuple
+								// we only want to increment the count once per unique destination,
+								// not once per connection, hence the flag and the check
+								if !uconnMap[srcDstKey].UPPSFlag {
+									for _, entry := range trustedAppReferenceList {
+										if (protocol == entry.protocol) && (dstPort == entry.port) {
+											if service != entry.service {
+												uconnMap[srcDstKey].UPPSFlag = true
+											}
+										}
+									}
+								}
+
+								// increment unique dst port: proto : service tuple list for host.
+								// Fine to do this with open connections as it's idempotent (yes, I like that word...)
+								if !stringInSlice(tuple, uconnMap[srcDstKey].Tuples) {
+									uconnMap[srcDstKey].Tuples = append(uconnMap[srcDstKey].Tuples, tuple)
+								}
+
+								// Check if invalid cert record was written before the uconns
+								// record, we'll need to update it with the tuples.
+								// Fine to do this with open connections as it's idempotent
+								if _, ok := certMap[dstKey]; ok {
+									// add tuple to invlaid cert list
+									if !stringInSlice(tuple, certMap[dstKey].Tuples) {
+										certMap[dstKey].Tuples = append(certMap[dstKey].Tuples, tuple)
+									}
+								}
+
+								// Replace existing duration if current duration is higher. It's fine
+								// to do this if the connection is still open as it will just get replaced
+								// later when the connection closes. If an open connection is responsible
+								// for the longest duration, we want to see that.
+								if duration > uconnMap[srcDstKey].MaxDuration {
+									uconnMap[srcDstKey].MaxDuration = duration
+								}
+
+								if duration > hostMap[srcKey].MaxDuration {
+									hostMap[srcKey].MaxDuration = duration
+								}
+								if duration > hostMap[dstKey].MaxDuration {
+									hostMap[dstKey].MaxDuration = duration
+								}
+
+								mutex.Unlock()
```

Comment:
Can we add a comment about not incrementing the connection counters until a connection is closed here?